week-3/todo-list/script.js

```diff
@@ -1,25 +1,33 @@
-function populateTodoList(todos) {
-  let list = document.getElementById("todo-list");
-  // Write your code to create todo list elements with completed and delete buttons here, all todos should display inside the "todo-list" element.
-}
-
-// These are the same todos that currently display in the HTML
-// You will want to remove the ones in the current HTML after you have created them using JavaScript
-let todos = [
-  { task: "Wash the dishes", completed: false },
-  { task: "Do the shopping", completed: false },
-];
-
-populateTodoList(todos);
+let list = document.getElementById("todo-list");
+const addTodo = document.querySelector("#add");
+const todos = [];
 
-// This function will take the value of the input field and add it as a new todo to the bottom of the todo list. These new todos will need the completed and delete buttons adding like normal.
-function addNewTodo(event) {
+function addNewTodo() {
   // The code below prevents the page from refreshing when we click the 'Add Todo' button.
   event.preventDefault();
-  // Write your code here... and remember to reset the input field to be blank after creating a todo!
+  const input = document.querySelector('input');
+  if (input.value != "") {
+    const li = document.createElement("li");
+    list.appendChild(li);
+    li.innerHTML = input.value;
+    li.classList.add('list-item');
+    const span = document.createElement("span");
+    const htmlContent = '<i id="check" onclick="this.parentNode.parentNode.style.textDecoration=`line-through`;" ></i>'
+      + '<i id="trash" onclick="this.parentNode.parentNode.remove()"></i>';
+    span.innerHTML = htmlContent;
+    li.appendChild(span);
+    todos.push(li);
+  }
 }
 
-// Advanced challenge: Write a fucntion that checks the todos in the todo list and deletes the completed ones (we can check which ones are completed by seeing if they have the line-through styling applied or not).
+populateTodoList(list);
+
+// Advanced challenge: Write a function that checks the todos in the todo list and deletes the completed ones (we can check which ones are completed by seeing if they have the line-through styling applied or not).
 function deleteAllCompletedTodos() {
   // Write your code here...
-}
+  for (const item of todos) {
+    if (item.getAttribute("style") != null) {
```

Comment:
This works, but feels a little indirect - whether an item has any style happens to work for your code right now, but I can imagine in the future someone may add some different style to add items, and then this button will delete all of them... Can you think how you could store/detect whether a todo is done more explicitly?

(This is a very advanced question, by the way, you'll learn more ideas around this in JS3!)

Reply:
Using `if (item.style.textDecoration)` explicitly checks for `text-decoration` instead of returning all inline styles as string using `item.getAttribute("style")`.